### PR TITLE
Add local.settings.json to gitignore

### PR DESCRIPTION
**Pull request by timbussmann**
a local.settings.json file can contain secrets/connection strings and therefore should never be committed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+local.settings.json
```
Comment (adamralph) [inline .gitignore:1]: Should this be under `# User-specific files` (line 45)?

Comment (adamralph) [reply .gitignore]: https://github.com/Particular/PlatformArchitecture/pull/506#discussion_r438676055 🙂 

Sure, let's keep the new line.

**Comment by timbussmann** [comment]
@adamralph it's a settings file which loads all strings of it's `Values` property into environment variables when running Azure Functions (and I think other Azure services too). It's used to contain connection strings when developing locally and to make sure they are not committed to source control. Azure Function project templates have this file in their .gitignore file as a default.

**Comment by adamralph** [comment]
OK, sounds legit. I'll approve.

**Pull request merged**